Status code can be 400, 401, 403 or .. !

rohe opened the pull request:
According to RFC6749 an OP shall return a 400 when it encounters an error in processing a request.
The OIDC core document in section 5.3.3 in the example uses 401.
During conformance testing I've also seen 403's in some cases.
So, there is reason to widen then scope for possible status_code denoting protocol errors.

The fact that I've seen software return a 200 with an error message is another matter and is already covered in the code.

tpazderka commented on src/oic/oic/__init__.py:
```diff
@@ -849,7 +849,7 @@ def do_user_info_request(self, method="POST", state="", scope="openid",
         elif resp.status_code == 500:
             raise PyoidcError("ERROR: Something went wrong: %s" % resp.text)
-        elif resp.status_code == 400:
+        elif 400 <= resp.status_code < 500 :
```
There is a space before the `:`.

Otherwise I am fine with that.

rohe commented:
Removed the space.
Also fixed a reference to a exception class that was wrong.
Actually the class in question is defined in 2 places.
Should fix that but that's the topic for another PR.
